src/utilities/geometry/Point3d.cpp

```diff
@@ -113,4 +113,25 @@ std::ostream& operator<<(std::ostream& os, const std::vector<Point3d>& pointVect
   return os;
 }
 
+// Writes a json string representing the vector of point vectors
+// Will deserialize in to a simple class that defines three double properties X,Y,Z
+std::ostream& operator<<(std::ostream& os, const std::vector<std::vector<Point3d>>& pointVectorVector) {
+  os << "[";
+  for (unsigned i = 0; i < pointVectorVector.size(); i++) {
+    os << "[";
+    Point3dVector points = pointVectorVector[i];
+    for (unsigned j = 0; j < points.size(); j++) {
+      os << "{";
+      os << "\"X\":" << points[j].x() << ",";
+      os << "\"Y\":" << points[j].y() << ",";
+      os << "\"Z\":" << points[j].z();
+      os << "},";
+    }
+    os << "],";
+  }
+  os << "]";
+
+  return os;
+}
```

Comment:
* Range-based for loop
* Point3d has an ostream operator already, can't we just use that already? At least I find uppercase X, Y, Z quite weird

Proposed implementation:

```
std::ostream& operator<<(std::ostream& os, const std::vector<std::vector<Point3d>>& pointVectorVector) {
  os << "[\n";
  for (const auto& points: pointVectorVector) {
    os << "  [";
    for (const auto& point: points) {
      os << "\n    " << point << ", ";
    }
    os << "\n  ],\n";
  }
  os << "]";

  return os;
}
```

And alternative relying on `<fmt/ranges.h>` and `<fmt/ostream.h>`

```c++
std::ostream& operator<<(std::ostream& os, const std::vector<std::vector<Point3d>>& pointVectorVector) {
  os << fmt::format("{}", pointVectorVector);
  return os;
}
```

Reply:
I'll make that change - the initial implementation was formatted so I could load the json file in to Simergy for debugging purposes so I could view the polygons output from doing intersections but on reflection it would be better to modify the back end code to accept an list of three doubles.

Done

Reply:
Thanks for the change. Note that i don't think there's anything wrong with producing a valid json for a purpose, I didn't have the context

Reply:
This was creating a json file I could load in to Simergy to see the output from intersections but I agree a more generic list of doubles is better so I used your implementation and changed our code to suit